### PR TITLE
Updated Vert.x core, Proton and MQTT server to 3.4.0 and needed Netty…

### PR DESCRIPTION
**Pull request by ppatierno**
… to 4.1.8.Final

Signed-off-by: ppatierno <ppatierno@live.com>

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -46,9 +46,9 @@
     <spring.version>4.3.2.RELEASE</spring.version>
     <spring-boot.version>1.4.0.RELEASE</spring-boot.version>
     <truth.version>0.28</truth.version>
-    <vertx.version>3.3.3</vertx.version>
-    <vertx-proton.version>3.4.0.Beta1</vertx-proton.version>
-    <vertx-mqtt.version>3.4.0.Beta1</vertx-mqtt.version>
+    <vertx.version>3.4.0</vertx.version>
+    <vertx-proton.version>3.4.0</vertx-proton.version>
```
Comment (sophokles73) [inline bom/pom.xml]: We do not have a CQ for vertx-proton 3.4.0. However, we do have an approved CQ for 3.4.0.Beta1 so we should revert to that version for the time being. And make sure to create a CQ for 3.4.0 ... :-)

File: bom/pom.xml [hunk 2]
```diff
@@ -46,9 +46,9 @@
...
+    <vertx-mqtt.version>3.4.0</vertx-mqtt.version>
```
Comment (sophokles73) [inline bom/pom.xml]: We could also revert to 3.4.0.Beta1 (for which we have an approved CQ) for the time being ...

**Comment by ppatierno** [comment]
@sophokles73 so I'm going to revert the PR to use 3.4.0.Beta1 for all Vert.x related components (core, proton, mqtt) and open a CQ for Vert.x Proton  3.4.0. Regarding Vert.x MQTT I updated the one we already have. Is that ok ? Finally we don't need a CQ for Vert.x core right ? because it's Eclipse project.
Btw, imho CQs are a good way for slowing down productivity :-)

**Comment by sophokles73** [comment]
I was more thinking of using 3.4.0 of vertx-core (for which we do not need a CQ) and use 3.4.0.Beta1 for vertx-proton and vertx-mqtt-server ...

CQs are also a good way of providing safety regarding IP related issues around using open source software. It is actually this process that enables us to engage in Eclipse projects at all. I know it's a PITA but we definitely need to go through it ...

**Comment by ppatierno** [comment]
@sophokles73 I committed and squashed with 3.4.0.Beta1 (only for mqtt server and proton) and opened this CQ (https://dev.eclipse.org/ipzilla/show_bug.cgi?id=12866) for Vert.x Proton 3.4.0.
All is still working fine :-)

**Comment by sophokles73** [comment]
Then I guess you should proceed and merge the PR ...

**Pull request merged**